### PR TITLE
[android-auto] New map deselection api

### PR DESCRIPTION
**Pull request by AndrewShkrob**
Better fix for https://github.com/organicmaps/organicmaps/pull/8789
Kudos to @strump

### CODE REVIEW

File: map/framework.cpp [hunk 1]
```diff
@@ -1923,6 +1923,12 @@ void Framework::DeactivateMapSelection()
   }
 }
 
+void Framework::DeactivateMapSelectionCircle()
+{
+    if (m_drapeEngine != nullptr)
+        m_drapeEngine->DeselectObject();
```
Comment (biodranik) [inline map/framework.cpp:1929]:
```suggestion
  if (m_drapeEngine)
    m_drapeEngine->DeselectObject();
  else
    LOG(LWARNING, "DeactivateMapSelectionCircle is called without initializing the Drape Engine");
```

**Review by biodranik — APPROVED** [review]
@AndrewShkrob thanks! A minor nit.

File: map/framework.cpp (hunk 1 above)
Comment (vng) [reply map/framework.cpp:1929]: No one will read this log :) Or otherwise we can pollute the whole Framework with it.

**Review by vng — APPROVED** [review]

File: map/framework.cpp (hunk 1 above)
Comment (biodranik) [reply map/framework.cpp:1929]: Would it be better to crash? Logs can be collected in theory using OM settings or ADB.

Comment (strump) [reply map/framework.cpp:1929]: We alredy have this `if (m_drapeEngine)` in `Framework::DeactivateMapSelection()`. I just copied it to a separate method.

Comment (vng) [reply map/framework.cpp:1929]: What crash? "If" is correct here and additional LOG gives you nothing.

Comment (biodranik) [reply map/framework.cpp:1929]: What is the correct case when Deactivate... method is called but drape is not initialized yet? Isn't it an abnormal situation?

**Pull request merged**